### PR TITLE
Optimize bitmap function count, use roaring cardinality method

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

optimize bitmap function count, use roaring cardinality method, this will more fast than current version,
for a simple benchmark 
```
#include <benchmark/benchmark.h>
#include <roaring/roaring.hh>
const int maxn=40960;
static roaring::Roaring r1;
static roaring::Roaring r2;
void init()
{
  for (int i = 0; i < maxn; ++i) {
    r1.add(rand());
    r2.add(rand());
  }
}

static void and_cardinality(benchmark::State& state) {
  long card =0;
  for (auto _ : state){
      card = r1.and_cardinality(r2);
  }
  std::string str=std::to_string(card);
}

static void cardinality_after_and(benchmark::State& state) {
  long card =0;
  for (auto _ : state){
      card = (r1&r2).cardinality();
  }
  std::string str=std::to_string(card);
}

BENCHMARK(and_cardinality)->Iterations(1000);
BENCHMARK(cardinality_after_and)->Iterations(1000);
// Run the benchmark
int main(int argc, char** argv)
{
   init();
   ::benchmark::Initialize(&argc, argv);
   ::benchmark::RunSpecifiedBenchmarks();
}
```
 
use `and_cardinality` is  about 3 to 4 times faster than first `and` the cal `cardinality`
```
--------------------------------------------------------------------------------
Benchmark                                      Time             CPU   Iterations
--------------------------------------------------------------------------------
and_cardinality/iterations:1000           796634 ns       794637 ns         1000
cardinality_after_and/iterations:1000    2926119 ns      2919778 ns         1000
```

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Do you have benchmark result for this improvement?

**Comment by yangzhg** [comment]
> Do you have benchmark result for this improvement?

attach to the comment

File: be/src/exprs/bitmap_function.cpp [hunk 1]
```diff
@@ -333,10 +333,10 @@ BigIntVal BitmapFunctions::bitmap_count(FunctionContext* ctx, const StringVal& s
     // zero size means the src input is a agg object
     if (src.len == 0) {
         auto bitmap = reinterpret_cast<BitmapValue*>(src.ptr);
-        return {bitmap->cardinality()};
+        return static_cast<int64_t>(bitmap->cardinality());
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply be/src/exprs/bitmap_function.cpp:336]: because cardinality() returns uint64_t

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**